src/fty_config_manager.cc

```diff
@@ -0,0 +1,448 @@
+
+            m_aug = aug_init(FILE_SEPARATOR, m_parameters.at(AUGEAS_LENS_PATH).c_str(), augeasOpt);
+            if (!m_aug)
+            {
+                throw ConfigurationException("Augeas tool initialization failed");
+            }
+
+            // Message bus init
+            m_msgBus = messagebus::MlmMessageBus(m_parameters.at(ENDPOINT_KEY), m_parameters.at(AGENT_NAME_KEY));
+            m_msgBus->connect();
+            
+            // Listen all incoming request
+            auto fct = std::bind(&ConfigurationManager::handleRequest, this, _1);
+            m_msgBus->receive(m_parameters.at(QUEUE_NAME_KEY), fct);
+        }
+        catch (messagebus::MessageBusException& ex)
+        {
+            log_error("Message bus error: %s", ex.what());
+        } catch (...)
+        {
+            log_error("Unexpected error: unknown");
+        }
+    }
+
+    /**
+     * Handle any request
+     * @param sender
+     * @param payload
+     * @return 
+     */
+    void ConfigurationManager::handleRequest(messagebus::Message msg)
+    {
+        try
+        {
+            log_debug("Configuration handle request");
+            
+            messagebus::UserData userData;
+            // Get request
+            messagebus::UserData data = msg.userData();
+            dto::config::ConfigQueryDto configQuery;
+            data >> configQuery;
+
+            log_debug("Config query-> action: %s, feature(s): %s", configQuery.action.c_str(), configQuery.featureName.c_str());
+            if (configQuery.action.size() == 0)
+            {
+                if ((configQuery.action.compare(SAVE_ACTION) == 0 && configQuery.featureName.size() == 0) ||
+                    (configQuery.action.compare(RESTORE_ACTION) == 0 && configQuery.data.size() == 0))
+                {
+                    throw ConfigurationException("Request not valid");
+                }
+            }
+            // Load augeas for any request (to avoid any cache).
+            aug_load(m_aug);
+            
+            // Check if the command is implemented
+            if (configQuery.action.compare(SAVE_ACTION) == 0)
+            {
+                // Response
+                dto::config::ConfigResponseDto respDto(configQuery.featureName, STATUS_FAILED);
+                std::map<std::string, cxxtools::SerializationInfo> configSiList;
+                // Split feature name 
+                std::vector<std::string> featureList = splitString(configQuery.featureName);
+                for(auto const& feature: featureList)
+                {
+                    // Get the configuration file path name from class variable m_parameters
+                    std::string configurationFileName = AUGEAS_FILES + m_parameters.at(feature) + ANY_NODES;
+                    log_debug("Configuration file name: %s", configurationFileName.c_str());
+                    cxxtools::SerializationInfo si;
+                    getConfigurationToJson(si, configurationFileName);
```

Comment:
Check if changing `getConfigurationToJson` to return a `cxxtools::SerializationInfo` object directly is possible.

Reply:
There are better approach for serialization in Json with cxxtools. The power of cxxtools with c++ is to overload operator=<<.

@MimiXav I could take a look with you to see if we could do it a better way in this case. It's not blocking point.